src/main/java/org/reaktivity/nukleus/http2/internal/routable/Correlation.java

```diff
@@ -19,25 +19,29 @@
 
 import java.util.Objects;
 
+import org.reaktivity.nukleus.http2.internal.routable.stream.SourceInputStreamFactory;
 import org.reaktivity.nukleus.http2.internal.router.RouteKind;
 
 public class Correlation
 {
     private final String source;
+    private final SourceInputStreamFactory.SourceInputStream sourceInputStream;
```

Comment:
Please don't let the internal `SourceInputStream` class escape outside of the `SourceInputStreamFactory` class.

Instead, please use a functional interface matching the signature of `doPromisedRequest` passed from source to target via `Correlation`.